adds tilt reagent

gy1ta23 opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You'll find a README and example file in .\html\changelogs\ for further instructions.

You can also find a template for adding your changelog directly to the PR description here: https://github.com/Baystation12/Baystation12/wiki/Automatic-changelog-generation
-->

Okay, we're going to come out of the gates honest with you, because I believe that you are reasonable people deserving of our respect. We've created space lean, and we called it tilt because that's a synonym for lean. You make it with cough medicine and space-up. It gets you pretty intoxicated, and if you drink too much you get real fucked up. That's all you really need.

Here's a picture:
![lean](https://user-images.githubusercontent.com/44277885/157798880-a9e2921b-667f-4d4d-ac80-6fa9e1f06422.png)


Credit to joeynosegay for his excellent idea, spritework, moral support and lean expertise.

🆑 
rscadd: Adds "tilt" - a downer drug made from mixing spaceup and cough medicine
/🆑

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+	glass_name = "tilt"
+	glass_desc = "The best way there is to get tilted."
+
+/datum/reagent/tilt/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
```
```suggestion
/datum/reagent/tilt/affect_blood(mob/living/carbon/M, alien, removed)
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+	glass_desc = "The best way there is to get tilted."
+
+/datum/reagent/tilt/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+	if(alien == IS_DIONA)
```
```suggestion
	if (alien == IS_DIONA)
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+		return
+
+	var/drug_strength = 10
+	if(alien == IS_SKRELL)
```
```suggestion
	if (alien == IS_SKRELL)
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+	M.add_chemical_effect(M.add_chemical_effect(CE_SLOWDOWN, 1))
+	M.slurring = max(M.slurring, 10)
+
+	if(prob(90))
```
```suggestion
	if (prob(90))
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+
+	if(prob(90))
+		M.drowsyness = max(M.drowsyness, 3)
+	if(prob(70))
```
```suggestion
	if (prob(70))
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+		M.drowsyness = max(M.drowsyness, 3)
+	if(prob(70))
+		M.confused = max(M.confused, 10)
+	if(prob(10))
```
```suggestion
	if (prob(10))
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+		M.emote("drool")
+		M.apply_effect(STUTTER, 3)
+
+/datum/reagent/tilt/overdose(var/mob/living/carbon/M, var/alien)
```
```suggestion
/datum/reagent/tilt/overdose(mob/living/carbon/M, alien)
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+
+/datum/reagent/tilt/overdose(var/mob/living/carbon/M, var/alien)
+	M.add_chemical_effect(CE_TOXIN, 1)
+	if(prob(70))
```
```suggestion
	if (prob(70))
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -744,6 +744,47 @@
+	M.add_chemical_effect(CE_TOXIN, 1)
+	if(prob(70))
+		M.add_chemical_effect(CE_BREATHLOSS, 5)
+	if(prob(10))
```
```suggestion
	if (prob(10))
```

Spookerton commented on code/modules/reagents/Chemistry-Recipes.dm:
```diff
@@ -137,6 +137,12 @@
+/datum/chemical_reaction/tilt
+	name = "Tilt"
+	result = /datum/reagent/tilt
+	required_reagents = list(/datum/reagent/antidexafen = 2, /datum/reagent/drink/space_up = 1)
```
```suggestion
	required_reagents = list(
		/datum/reagent/antidexafen = 2,
		/datum/reagent/drink/space_up = 1
	)
```

Spookerton reviewed (changes requested):
Basic style stuff. Going to take a quick look at the effect strength too.

Spookerton reviewed (dismissed):
A single 5u sip is down and out for as long as it takes to process all the way. Needs to scale on chem_doses[type] instead of current behavior.

gy1ta23 commented:
style changes committed, chem_doses behavior added in what I believe represents what you wanted

gy1ta23 committed:
adds tilt


Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/Chemistry-Recipes.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
fixes effects


ACTUALLY FIXES STYLE STUFF

SierraKomodo commented:
Wiki pages that will need updating:
- https://bay.ss13.me/en/Guides/Chemistry

Spookerton commented:
updooted
